exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentContextImpl.java

```diff
@@ -362,12 +368,35 @@ public boolean isUserAuthenticationEnabled() {
 
   @Override
   public void addRuntimeFilter(RuntimeFilterWritable runtimeFilter) {
-    this.runtimeFilterSink.aggregate(runtimeFilter);
+    this.runtimeFilterWritable = runtimeFilter;
+    if (enableRFWaiting) {
+      condition.signal();
+    }
```

Comment:
We can add the runtimeFilterWritable under `lock()` to prevent any race condition although there is no harm with this race. But there is no issue in adding it under lock either that way its clear that lock and condition variable is for `runtimeFilterWritable`.

```
lockForRF.lock()
try () {
     this.runtimeFilterWritable = runtimeFilter;
     if (enableRFWaiting) {
          // wake up the waiting thread in RuntimeFilterRecordBatch
          conditionForRF.signal();
     }
} finally {
    lockForRF.unlock()
}
```

Reply:
Here there's a usage bug about the signal ,I have fixed that on the latest commit. But I don't want to add unnecessary lock to the runtimeFilterWritable var. Drill's code has lots of sync usage. I always try to thin the block size, though it maybe introduce bugs. If you insist on this, I am open to do it at another commit. Thanks!